src/libs/ui/docsetsdialog.cpp

```diff
@@ -702,6 +706,29 @@ QNetworkReply *DocsetsDialog::download(const QUrl &url)
     return reply;
 }
 
+void DocsetsDialog::cancelDownload(const QModelIndex &index)
+{
+    // Find and delete download jobs corresponding to index
+    for (QNetworkReply *reply : m_replies) {
+        if (reply->property(ListItemIndexProperty).toInt() == index.row()
```

Comment:
It's better to return as early as possible, and avoid large nested blocks. In this case simple invert the condition and and use `continue`.